src/mugs.js

```diff
@@ -1,3 +1,37 @@
+/**
+ * A mug is a question, containing data, bind, and control elements.
+ *
+ * Main Properties
+ *  nodeID
+ *  label
+ *  readOnlyControl
+ *  itemsetData
+ *  imageSize: Images only. Options to reduce image size before sending form.
+ *
+ * Data Source Properties
+ *
+ * Media Properties
+ *  mediaItext: Multimedia (image, audio, video, inline video) attached to the question.
+ *
+ * Logic Properties
+ *  calculateAttr: Hidden questions only. The calculation that generates the question's value.
+ *  requiredAttr: Boolean. Whether or not user must enter a value for the question.
+ *  relevantAttr: Boolean expression that determines whether or not to display a question to the end user.
+ *  constraintAttr: Boolean expression that, if false, will prevent the user from proceeding past the question.
+ *  repeat_count: Repeat groups only. An integer expression that determines the number of groups to generate.
+ *
+ * Advanced Properties
+ *  dataSource
+ *  dataValue
+ *  defaultValue: An expression that will be assigned to a question before/until a user changes the value.
```

Comment:
Can you include the distinction between `dataValue` and `defaultValue` ? One is inserted into the data instance (I think `dataValue`, the other is a setValue (I think `defaultValue`). Also the one that inserts into the data instance deprecated